File: VB/Dennis.Spelling.Win/SpellCheckerViewController.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.XtraSpellChecker
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.ExpressApp.Win.Editors

Namespace Dennis.Spelling.Win
	Public Class SpellCheckerViewController
		Inherits ViewController
		Private spellCheckerCore As SpellChecker
		Private checkActionCore As SimpleAction
		Private parentControlCore As Control
		Public Sub New()
			checkActionCore = New SimpleAction(Me, "Check", PredefinedCategory.RecordEdit)
			checkActionCore.ToolTip = "Check Spelling"
			checkActionCore.Caption = checkActionCore.ToolTip
			checkActionCore.Category = "RecordEdit"
			checkActionCore.ImageName = "BO_Task"
			checkActionCore.TargetViewType = ViewType.DetailView
			AddHandler checkActionCore.Execute, AddressOf checkActionCore_Execute
		End Sub
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			If View.IsRoot Then
				parentControlCore = CType(View.Control, Control)
			End If
			SetupSpellChecker()
			SetupGridView()
		End Sub
		Protected Overridable Sub SetupSpellChecker()
			spellCheckerCore = New SpellChecker()
			SpellCheckerInitializer.Initialize(spellCheckerCore)
		End Sub
		Protected Overridable Sub SetupGridView()
			If TypeOf View Is DevExpress.ExpressApp.ListView Then
				Dim lv As DevExpress.ExpressApp.ListView = CType(View, DevExpress.ExpressApp.ListView)
				Dim gridListEditor As GridListEditor = TryCast(lv.Editor, GridListEditor)
				If gridListEditor IsNot Nothing Then
					RemoveHandler gridListEditor.GridView.ShownEditor, AddressOf GridView_ShownEditor
					AddHandler gridListEditor.GridView.ShownEditor, AddressOf GridView_ShownEditor
				End If
			End If
		End Sub
		Private Sub GridView_ShownEditor(ByVal sender As Object, ByVal e As EventArgs)
			If spellCheckerCore IsNot Nothing Then
				Dim activeControl As Control = (CType(sender, GridView)).ActiveEditor
				spellCheckerCore.SetShowSpellCheckMenu(activeControl, True)
				If spellCheckerCore.SpellCheckMode = SpellCheckMode.AsYouType Then
					spellCheckerCore.Check(activeControl)
				End If
			End If
		End Sub
		Private Sub checkActionCore_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
			If spellCheckerCore.ParentContainer Is Nothing Then
				spellCheckerCore.ParentContainer = parentControlCore
			End If
			spellCheckerCore.CheckContainer()
		End Sub
		Public ReadOnly Property SpellChecker() As SpellChecker
			Get
				Return spellCheckerCore
			End Get
		End Property
		Public ReadOnly Property CheckAction() As SimpleAction
			Get
				Return checkActionCore
			End Get
		End Property
	End Class
End Namespace

File: CS/Dennis.Spelling.Win/SpellCheckerViewController.cs
using System;
using System.Windows.Forms;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.XtraSpellChecker;
using DevExpress.ExpressApp.Actions;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.ExpressApp.Win.Editors;

namespace Dennis.Spelling.Win {
    public class SpellCheckerViewController : ViewController {
        private SpellChecker spellCheckerCore;
        private SimpleAction checkActionCore;
        private Control parentControlCore;
        public SpellCheckerViewController() {
            checkActionCore = new SimpleAction(this, "Check", PredefinedCategory.RecordEdit);
            checkActionCore.Caption = checkActionCore.ToolTip = "Check Spelling";
            checkActionCore.Category = "RecordEdit";
            checkActionCore.ImageName = "BO_Task";
            checkActionCore.TargetViewType = ViewType.DetailView;
            checkActionCore.Execute += checkActionCore_Execute;
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            if (View.IsRoot)
                parentControlCore = (Control)View.Control;
            SetupSpellChecker();
            SetupGridView();
        }
        protected virtual void SetupSpellChecker() {
            spellCheckerCore = new SpellChecker();
            SpellCheckerInitializer.Initialize(spellCheckerCore);
        }
        protected virtual void SetupGridView() {
            if (View is DevExpress.ExpressApp.ListView) {
                DevExpress.ExpressApp.ListView lv = (DevExpress.ExpressApp.ListView)View;
                GridListEditor gridListEditor = lv.Editor as GridListEditor;
                if (gridListEditor != null) {
                    gridListEditor.GridView.ShownEditor -= GridView_ShownEditor;
                    gridListEditor.GridView.ShownEditor += GridView_ShownEditor;
                }
            }
        }
        private void GridView_ShownEditor(object sender, EventArgs e) {
            if (spellCheckerCore != null) {
                Control activeControl = ((GridView)sender).ActiveEditor;
                spellCheckerCore.SetShowSpellCheckMenu(activeControl, true);
                if (spellCheckerCore.SpellCheckMode == SpellCheckMode.AsYouType)
                    spellCheckerCore.Check(activeControl);
            }
        }
        private void checkActionCore_Execute(object sender, SimpleActionExecuteEventArgs e) {
            if (spellCheckerCore.ParentContainer == null)
                spellCheckerCore.ParentContainer = parentControlCore;
            spellCheckerCore.CheckContainer();
        }
        public SpellChecker SpellChecker {
            get { return spellCheckerCore; }
        }
        public SimpleAction CheckAction {
            get { return checkActionCore; }
        }
    }
}